src/pages/dapp/browser.js

```diff
@@ -308,7 +310,12 @@ class DAppBrowser extends Component {
         case 'eth_sendTransaction': {
           callAuthVerify(async () => {
             try {
-              const nonce = await this.provider.getTransactionCount(address);
+              let nonce = await this.provider.getTransactionCount(address);
```

Comment:
No, we shouldn't need to save a local nonce value. The rsk3.js library can get the total count of confirmed and pending. Rwallet-server is using that function. It's the **pending** param. Try it out.

const nonce = await this.provider. getTransactionCount(address, 'pending');